### ISSUE TITLE
Wrong install path ?

### ISSUE THREAD

**Issue by nicogodet** [conda-forge/libmed-feedstock#47]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Based on https://docs.conda.io/projects/conda-build/en/stable/user-guide/environment-variables.html, `LIBRARY_PREFIX` should point to `<build prefix>\Library`.
So med files should be installed in `<build prefix>\Library\[include|bin|lib]` but they are installed in `<build prefix>\[include|bin]`.

In package that needs libmed, I need to include `-I%CONDA_PREFIX%\include` to find `med.hf`. If installed in Library\include, I wouldn't have to specify include folder.

### Installed packages

```shell
# packages in environment at D:\miniforge3\envs\build-telemac:
#
# Name                    Version                   Build  Channel
blas                      1.0                         mkl
brotli                    1.0.9                h2bbff1b_7
brotli-bin                1.0.9                h2bbff1b_7
brotlipy                  0.7.0           py310h2bbff1b_1002
bzip2                     1.0.8                he774522_0
ca-certificates           2023.08.22           haa95532_0
certifi                   2023.7.22       py310haa95532_0
cffi                      1.15.1          py310h2bbff1b_3
charset-normalizer        2.0.4              pyhd3eb1b0_0
contourpy                 1.0.5           py310h59b6b97_0
cryptography              41.0.3          py310h89fc84f_0
cycler                    0.11.0             pyhd3eb1b0_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.12.1               ha860e81_0
giflib                    5.2.1                h8cc25b3_3
glib                      2.69.1               h5dc1a3c_2
hdf5                      1.10.6               h1756f20_1
icc_rt                    2022.1.0             h6049295_2
icu                       58.2                 ha925a31_3
idna                      3.4             py310haa95532_0
intel-openmp              2023.1.0         h59b6b97_46319
jpeg                      9e                   h2bbff1b_1
kiwisolver                1.4.4           py310hd77b12b_0
krb5                      1.20.1               h5b6d351_0
lerc                      3.0                  hd77b12b_0
libbrotlicommon           1.0.9                h2bbff1b_7
libbrotlidec              1.0.9                h2bbff1b_7
libbrotlienc              1.0.9                h2bbff1b_7
libclang                  14.0.6          default_hb5a9fac_1
libclang13                14.0.6          default_h8e68704_1
libdeflate                1.17                 h2bbff1b_1
libffi                    3.4.4                hd77b12b_0
libflang                  5.0.0           h6538335_20180525    conda-forge
libiconv                  1.16                 h2bbff1b_2
libmed                    4.1.1           py310h6f30c0e_3    conda-forge
libpng                    1.6.39               h8cc25b3_0
libpq                     12.15                h906ac69_1
libsqlite                 3.43.2               hcfcfb64_0    conda-forge
libtiff                   4.5.1                hd77b12b_0
libwebp                   1.3.2                hbc33d0d_0
libwebp-base              1.3.2                h2bbff1b_0
libxml2                   2.10.4               h0ad7f3c_1
libxslt                   1.1.37               h2bbff1b_1
libzlib                   1.2.13               hcfcfb64_5    conda-forge
llvm-meta                 5.0.0                         0    conda-forge
lz4-c                     1.9.4                h2bbff1b_0
m2w64-binutils            2.25.1                        5
m2w64-bzip2               1.0.6                         6
m2w64-crt-git             5.0.0.4636.2595836               2
m2w64-gcc                 5.3.0                         6
m2w64-gcc-ada             5.3.0                         6
m2w64-gcc-fortran         5.3.0                         6
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gcc-objc            5.3.0                         6
m2w64-gmp                 6.1.0                         2
m2w64-headers-git         5.0.0.4636.c0ad18a               2
m2w64-isl                 0.16.1                        2
m2w64-libiconv            1.14                          6
m2w64-libmangle-git       5.0.0.4509.2e5a9a2               2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
m2w64-make                4.1.2351.a80a8b8               2
m2w64-metis               5.1.0                         1
m2w64-mpc                 1.0.3                         3
m2w64-mpfr                3.1.4                         4
m2w64-msmpi               10.1.1               h82bb817_5    nicogodet
m2w64-mumps               5.2.1                         0    nicogodet
m2w64-pkg-config          0.29.1                        2
m2w64-scalapack           2.2.1.6                       0    nicogodet
m2w64-toolchain           5.3.0                         7
m2w64-tools-git           5.0.0.4592.90b8472               2
m2w64-windows-default-manifest 6.4                           3
m2w64-winpthreads-git     5.0.0.4634.697f757               2
m2w64-zlib                1.2.8                        10
matplotlib                3.7.2           py310haa95532_0
matplotlib-base           3.7.2           py310h4ed8f06_0
mkl                       2023.1.0         h6b88ed4_46357
mkl-service               2.4.0           py310h2bbff1b_1
mkl_fft                   1.3.8           py310h2bbff1b_0
mkl_random                1.2.4           py310h59b6b97_0
msys2-conda-epoch         20160418                      1
munkres                   1.1.4                      py_0
numpy                     1.21.5          py310h85e1a82_4
numpy-base                1.21.5          py310hb5c95e7_4
openjpeg                  2.4.0                h4fc8c34_0
openmp                    5.0.0                    vc14_1    conda-forge
openssl                   3.1.3                hcfcfb64_0    conda-forge
packaging                 23.1            py310haa95532_0
pcre                      8.45                 hd77b12b_0
pillow                    10.0.1          py310h045eedc_0
pip                       23.3            py310haa95532_0
platformdirs              3.10.0          py310haa95532_0
ply                       3.11            py310haa95532_0
pooch                     1.7.0           py310haa95532_0
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 23.2.0          py310haa95532_0
pyparsing                 3.0.9           py310haa95532_0
pyqt                      5.15.7          py310hd77b12b_0
pyqt5-sip                 12.11.0         py310hd77b12b_0
pysocks                   1.7.1           py310haa95532_0
python                    3.10.12         h4de0772_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd3eb1b0_0
python_abi                3.10                    4_cp310    conda-forge
qt-main                   5.15.2               h879a1e9_9
qt-webengine              5.15.9               h5bd16bc_7
qtwebkit                  5.212                h2bbfb41_5
requests                  2.31.0          py310haa95532_0
scipy                     1.10.1          py310h309d312_1
setuptools                68.0.0          py310haa95532_0
sip                       6.6.2           py310hd77b12b_0
six                       1.16.0             pyhd3eb1b0_1
sqlite                    3.41.2               h2bbff1b_0
tbb                       2021.8.0             h59b6b97_0
tk                        8.6.12               h2bbff1b_0
toml                      0.10.2             pyhd3eb1b0_0
tornado                   6.3.3           py310h2bbff1b_0
tzdata                    2023c                h04d1e81_0
ucrt                      10.0.20348.0         haa95532_0
urllib3                   1.26.16         py310haa95532_0
vc                        14.2                 h21ff451_1
vc14_runtime              14.36.32532         hdcecf7f_17    conda-forge
vs2015_runtime            14.36.32532         h05e6639_17    conda-forge
wheel                     0.41.2          py310haa95532_0
win_inet_pton             1.1.0           py310haa95532_0
xz                        5.4.2                h8cc25b3_0
zlib                      1.2.13               hcfcfb64_5    conda-forge
zstd                      1.5.5                hd43e919_0
mamba
```

### Environment info

```shell
__    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (1.4.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


     active environment : build-telemac
    active env location : D:\miniforge3\envs\build-telemac
            shell level : 2
       user config file : C:\Users\godet\.condarc
 populated config files : D:\miniforge3\.condarc
                          C:\Users\godet\.condarc
          conda version : 23.3.1
    conda-build version : 3.26.1
         python version : 3.10.12.final.0
       virtual packages : __archspec=1=x86_64
                          __win=0=0
       base environment : D:\miniforge3  (writable)
      conda av data dir : D:\miniforge3\etc\conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/nicogodet/win-64
                          https://conda.anaconda.org/nicogodet/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
                          https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : D:\miniforge3\pkgs
                          C:\Users\godet\.conda\pkgs
                          C:\Users\godet\AppData\Local\conda\conda\pkgs
       envs directories : D:\miniforge3\envs
                          C:\Users\godet\.conda\envs
                          C:\Users\godet\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/23.3.1 requests/2.31.0 CPython/3.10.12 Windows/10 Windows/10.0.19045
          administrator : False
             netrc file : None
           offline mode : False
```

**Comment by ldallolio** [comment]
Hi Nicolas,

I agree with you, but I do not really understand what should be changed : the [bld.bat](https://github.com/conda-forge/libmed-feedstock/blob/1a3edb25be3f68992476adca28e1408c6cc5b8a6/recipe/bld.bat#L37) file seems to already specify exactly what you are suggesting (or not?) : 

`cmake -G "Ninja" -S %SRC_DIR%^
    -Wno-dev ^
    -D CMAKE_BUILD_TYPE=%BUILD_TYPE% ^
    -D CMAKE_INSTALL_PREFIX:FILEPATH=%LIBRARY_PREFIX% ^
    -D CMAKE_PREFIX_PATH:FILEPATH=%LIBRARY_PREFIX% ^
    -D CMAKE_INCLUDE_PATH:FILEPATH=%LIBRARY_PREFIX%\include ^`

If you could please help me understand what's wrong here I will gladly fix it, here is what I can see in the CMakeLists.txt in the sources :

`SET(CONF_INCLUDE_DIRS "${CMAKE_INSTALL_PREFIX}/${INSTALL_INCLUDE_DIR}")
CONFIGURE_PACKAGE_CONFIG_FILE(config/cmake_files/${PROJECT_NAME}Config.cmake.in 
    ${PROJECT_BINARY_DIR}/to_install/${PROJECT_NAME}Config.cmake
    INSTALL_DESTINATION "${INSTALL_CMAKE_CONFIG_DIR}"
    PATH_VARS CONF_INCLUDE_DIRS INSTALL_CMAKE_CONFIG_DIR CMAKE_INSTALL_PREFIX HDF5_ROOT_DIR MPI_ROOT_DIR)`
    
I will keep this issue open until I found the real reason behind this, thank you for pointing this out,

All the best,

**Comment by nicogodet** [comment]
Hi, 

Indeed, bld.bat seems fine.
I try to replicate the build locally without success (build fails with both ninja and mingw) and I don't have much time to dig the issue.

**Comment by ldallolio** [comment]
Hi Nicolas,

I think I get what's happening : windows builds seem to be broken because
the Salome site seems to have blacklisted the conda forge build environment
(probably because it downloads many times the med package when building
variants), I can see the HTTP 403 errors in the logs. I will try to point
to another archive and see if it helps

Thank you for looking into this, have a nice day!
Best regards,
Luca

Il giorno mar 24 ott 2023 alle ore 08:28 Nicolas Godet <
***@***.***> ha scritto:

**Comment by nicogodet** [comment]
Yup, I saw your PR. I raise the issue to a developer who initially provide the new address for medfile archive. I will keep in touch.

For context, I try to build a conda package of openTelemac-Mascaret suite: 

- http://www.opentelemac.org/
- https://github.com/nicogodet/telemac-mascaret-feedstock

**Comment by ldallolio** [comment]
Hi Nicolas,

A quick word to let you know what's happening : 
- we worked around the repository blacklist issue by creating a copy (it would be better to make the official one working with conda forge because the conda bots would be able to pick new versions automatically), while we wait for conda forge to be whitelisted (if it will ever happen)
- I disabled again windows build : I think it is blocking now because hdf5 does not seem to have a fortran version for windows, will look into that in the future
All the best

**Comment by nicogodet** [comment]
Hi, 

I've been working on my side to fix hdf and med package
https://github.com/nicogodet/hdf5-otm-feedstock
https://github.com/nicogodet/med-otm-feedstock

**Comment by nicogodet** [comment]
I wonder how an old but fixed version of hdf5 could be pushed to conda-forge

**Comment by ldallolio** [comment]
My advice would be to work on a pull request, I prepared one for the same reason (not working yet) :
https://github.com/conda-forge/hdf5-feedstock/pull/203
the old version would not be an issue I think : conda forge uses branches for this :
https://conda-forge.org/docs/maintainer/updating_pkgs.html#maintaining-several-versions

**Comment by nicogodet** [comment]
Will be fixed with #50
